packages/core/src/textures/BaseTexture.js

```diff
@@ -632,4 +639,29 @@ export default class BaseTexture extends EventEmitter
 
         return null;
     }
+
+    /**
+     * Called from multi-texture rendered to get a new batch number.
+     *
+     * @static
+     * @return {number} new texture batch number
+     */
+    static _batchBegin()
+    {
+        return ++BaseTexture._globalBatch;
+    }
+
+    /**
+     * Called from multi-texture rendered to save the last batch number.
+     * Only first 32 bits matter.
+     *
+     * @static
+     * @param {number} savedValue last used batch number
+     */
+    static _batchEnd(savedValue)
+    {
+        BaseTexture._globalBatch = savedValue | 0;
```

Comment:
looking good, can you explain how this magic line works pls? Why only first 32 bits matter? Just curious! :)

Reply:
just in case we have more than 2^54 texture batches. Ugh... 100 years I think? lets remove that.

Reply:
ok, changed.